src/main/java/org/testng/xml/internal/TestNamesMatcher.java

```diff
@@ -0,0 +1,106 @@
+package org.testng.xml.internal;
+
+import java.util.Iterator;
+import java.util.List;
+import org.testng.TestNGException;
+import org.testng.collections.Lists;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+
+/**
+ * The class to work with "-testnames"
+ *
+ */
+public final class TestNamesMatcher {
+    
+    private final List<XmlSuite> cloneSuites = Lists.newArrayList();
+    private final List<String> matchedTestNames = Lists.newArrayList();
+    private final List<XmlTest> matchedTests = Lists.newArrayList();
+    private List<String> testNames = Lists.newArrayList();
```

Comment:
This initialization is not needed, since `testNames` is always being initialized only via the constructor.

Reply:
And could be final

Reply:
final field has to be initialized, otherwise compile error.

Reply:
testNames can't be set as final, other wise constructor compile error.

Reply:
Exactly: set it final and init it only in the constructor